File: GXpert/GXpert.Web/Modules/Schools/SchoolClass/SchoolClassColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Schools.Columns;

[ColumnsScript("Schools.SchoolClass")]
[BasedOnRow(typeof(SchoolClassRow), CheckNames = true)]
public class SchoolClassColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ClassTitle { get; set; }
    [EditLink]
    public string Division { get; set; }
    public string MediumTitle { get; set; }
    public string ClassTeacherPrn { get; set; }
    public string Title { get; set; }
    public string AcademicYearName { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Schools/SchoolClassColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SchoolClassRow } from "./SchoolClassRow";

export interface SchoolClassColumns {
    Id: Column<SchoolClassRow>;
    ClassTitle: Column<SchoolClassRow>;
    Division: Column<SchoolClassRow>;
    MediumTitle: Column<SchoolClassRow>;
    ClassTeacherPrn: Column<SchoolClassRow>;
    Title: Column<SchoolClassRow>;
    AcademicYearName: Column<SchoolClassRow>;
}

export class SchoolClassColumns extends ColumnsBase<SchoolClassRow> {
    static readonly columnsKey = 'Schools.SchoolClass';
    static readonly Fields = fieldsProxy<SchoolClassColumns>();
}